### PR TITLE
chore: rename hermes-inspector-msggen to @react-native/hermes-inspector-msggen

### PR DESCRIPTION
**Pull request by gabrieldonadel**
## Summary

This PR renames ` hermes-inspector-msggen` to `@react-native/hermes-inspector-msggen`, moves it to  `packages/hermes-inspector-msggen` and makes the package private  as part of RFC480.

Related to https://github.com/facebook/react-native/issues/34692

## Changelog
 

[General] [Changed] - Move and rename `hermes-inspector-msggen` to `@react-native/hermes-inspector-msggen`

## Test Plan

run `sh ReactCommon/hermes/inspector/tools/run_msggen`

### CODE REVIEW

**Review by cortinico — CHANGES_REQUESTED** [review]
Can you remove the `yarn.lock` file?

**Comment by gabrieldonadel** [comment]
> Can you remove the `yarn.lock` file?

Sure, done

File: ReactCommon/hermes/inspector/tools/run_msggen [hunk 1]
```diff
@@ -12,7 +12,7 @@ FBSOURCE=$(hg root)
 CLANG_FORMAT="${FBSOURCE}/tools/third-party/clang-format/clang-format"
 SIGNEDSOURCE="${FBSOURCE}/tools/signedsource"
 
-cd "${DIR}/msggen"
+cd "${DIR}/../../../../packages/hermes-inspector-msggen"
```
Comment (cortinico) [inline ReactCommon/hermes/inspector/tools/run_msggen:15]: Can we `pushd` and `popd` here as the paths are so distant?

Comment (gabrieldonadel) [reply ReactCommon/hermes/inspector/tools/run_msggen]: Sure, I've just pushed a commit updating this

File: ReactCommon/hermes/inspector/tools/run_msggen [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 set -e
 
+ROOT=$(pwd)
+pushd $ROOT >/dev/null || exit
```
Comment (cortinico) [inline ReactCommon/hermes/inspector/tools/run_msggen:6]: You don't need `|| exit` as you have a `set -e` on top

File: ReactCommon/hermes/inspector/tools/run_msggen [hunk 3]
```diff
@@ -12,7 +15,8 @@ FBSOURCE=$(hg root)
 CLANG_FORMAT="${FBSOURCE}/tools/third-party/clang-format/clang-format"
 SIGNEDSOURCE="${FBSOURCE}/tools/signedsource"
 
-cd "${DIR}/msggen"
+popd >/dev/null || exit
+cd "packages/hermes-inspector-msggen"
```
Comment (cortinico) [inline ReactCommon/hermes/inspector/tools/run_msggen:19]: I would suggest we `pushd "${DIR}/msggen"` (or `$ROOT/packages/hermes-inspector-msggen`) and `popd` at the end of this script so the user is back in the folder where the script was invoked.

**Review by cortinico — COMMENTED** [review]
Can you address the comments and rebase?

File: ReactCommon/hermes/inspector/tools/run_msggen (hunk 2 above)
Comment (gabrieldonadel) [reply ReactCommon/hermes/inspector/tools/run_msggen:6]: Fixed

File: ReactCommon/hermes/inspector/tools/run_msggen (hunk 3 above)
Comment (gabrieldonadel) [reply ReactCommon/hermes/inspector/tools/run_msggen]: Sure, updated

File: packages/hermes-inspector-msggen/package.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
-  "name": "hermes-inspector-msggen",
+  "name": "@react-native/hermes-inspector-msggen",
+  "private": true,
   "version": "1.0.0",
```
Comment (cortinico) [inline packages/hermes-inspector-msggen/package.json:4]: This should be 0.71.0

**Review by cortinico — CHANGES_REQUESTED** [review]

File: packages/hermes-inspector-msggen/package.json (hunk 4 above)
Comment (gabrieldonadel) [reply packages/hermes-inspector-msggen/package.json]: Fixed

**Pull request closed without merging**